lebedigital/openbis/interbis.py

```diff
@@ -0,0 +1,495 @@
+        props_list = list(self.get_sample_type(sample_type)
+                          .get_property_assignments()
+                          .df['propertyType'])
+
+        df = pd.DataFrame()
+        # Getting the metadata of every entry in props_list
+        for prop in props_list:
+            pt = self.get_property_type(prop)
+            prop_series = pd.Series(pt.attrs.all())
+            prop_df = pd.DataFrame(
+                {prop: prop_series.values}, index=prop_series.index)
+
+            # Combining the props together into a dataframe
+            if df.empty:
+                df = prop_df
+            else:
+                df = df.join(prop_df)
+
+        df = df.transpose().reset_index(drop=True)
+        df = df.fillna(value="")
+        return df
+
+    def get_sample_dict(self, identifier: str) -> dict:
+        """Fetches a dictionary filled with information about the sample
+
+        Very useful method actually if you already have the identifier. You can get all information about
+        the sample from it like all metadata, collection, type, etc.
+
+        Args:
+            identifier (str): Identifier of the sample (identifier or permID)
+
+        Returns:
+            dict: dict containing all information about the sample
+        """
+        sample = self.get_sample(identifier)
+        sample_info_dict = sample.attrs.all()
+        sample_prop_dict = {key.upper(): val for key, val in sample.p.all().items()}
+        full_dict = (sample_info_dict | sample_prop_dict)
+
+        # copying the information from key experiment to collection for ease of use
+        full_dict['collection'] = full_dict.get('experiment')
+        return full_dict
+
+    def get_sample_identifier(self, name: str) -> str:
+        """
+
+        Returns the full identifier of the sample
+
+        Args:
+            name (str): '$name' attribute of the sample
+
+        Returns:
+            str: Identifier of sample
+        """
+        sample_df = self.get_samples(where={'$name': name}).df
+        if len(sample_df.index) > 1 or len(sample_df.index) == 0:
+            raise ValueError(
+                f'Could not find unique sample, the amount of samples with that name is {len(sample_df.index)}')
+
+        return sample_df['identifier'].values[0]
+
+    def get_dataset_identifier(self, name: str) -> list[str]:
+        """
+
+        Returns the permId of the dataset
+
+        Args:
+            name (str): '$name' attribute of the dataset
+
+        Returns:
+            list[str]: List of the dataset identifiers
+        """
+        df = self.get_datasets(**{"$NAME": name}).df
+        if len(df) == 0:
+            logging.warning("No dataset found")
+            return []
+        elif len(df) > 1:
+            logging.warning(f"More than one dataset with the name {name} found")
+
+        return df['permId'].to_list()
+
+    def get_collection_identifier(self, collection_code: str) -> str:
```

Comment:
Here it is the case that code != name. The code of the collection is the last part of the unique identifier (SPACE/PROJECT/COLLECTION). The method is there because more often than not there is a need to use the full Identifier of the collection instead of just the code.

Here the (props='$name') part was unnecessary so i deleted it.